keylime/cloud_verifier_tornado.py

```diff
@@ -856,6 +857,53 @@ async def invoke_provide_v(agent):
         asyncio.ensure_future(process_agent(agent, states.GET_QUOTE))
 
 
+async def invoke_notify_error(agent, tosend):
+    if agent is None:
+        raise Exception("Agent deleted while being processed")
+    res = tornado_requests.request(
+        "POST",
+        f"http://{agent['ip']}:{agent['port']}/v{agent['supported_version']}/notifications/revocation",
+        data=tosend,
+        context=agent["ssl_context"],
+    )
+    response = await res
+
+    if response.status_code != 200:
+        logger.warning(
+            "Unexpected Notify Revocation response error for cloud agent %s, Error: %s",
+            agent["agent_id"],
+            response.status_code,
+        )
+
+
+def notify_error_worker(agent, tosend, loop):
+    asyncio.set_event_loop(loop)
+    loop.run_until_complete(invoke_notify_error(agent, tosend))
+
+
+def notify_error(agent, msgtype="revocation", event=None):
+    notifiers = revocation_notifier.get_notifiers()
+    if len(notifiers) == 0:
+        return
+
+    tosend = cloud_verifier_common.prepare_error(agent, msgtype, event)
+    if "webhook" in notifiers:
+        revocation_notifier.notify_webhook(tosend)
+    if "zeromq" in notifiers:
+        revocation_notifier.notify(tosend)
+    if "agent" in notifiers:
+        loop = asyncio.new_event_loop()
```

Comment:
Why is the loop created here and not in the worker?

Reply:
Fixed.